### PR TITLE
update(cmd): bumped oras to v.2.4.0.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

**What this PR does / why we need it**:

See oras changelog: https://github.com/oras-project/oras-go/releases/tag/v2.4.0.
We already use:
* github.com/opencontainers/image-spec v1.1.0

as requested by the breaking changes in oras 2.4.0.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/registry/pull/pull_test.go [hunk 1]
```diff
@@ -297,7 +297,7 @@ var registryPullTests = Describe("pull", func() {
 				args = []string{registryCmd, pullCmd, newReg, "--plain-http", "--config", configFile}
 			})
 			pullAssertFailedBehavior(registryPullUsage, fmt.Sprintf("ERROR unable to create new repository with ref %s: "+
-				"invalid reference: invalid digest; invalid checksum digest format\n", newReg))
+				"invalid reference: invalid digest %q: invalid checksum digest format\n", newReg, "something"))
```
Comment (cpanato) [inline cmd/registry/pull/pull_test.go:300]: is this correct? why not add the hard coded var straight in the string?

Comment (FedeDP) [reply cmd/registry/pull/pull_test.go:300]: Just a preference to let `Sprintf` quote the string for me.

Comment (cpanato) [reply cmd/registry/pull/pull_test.go:300]: humm, ok

**Review by alacuku — APPROVED** [review]
/LGTM

**Review by leogr — APPROVED** [review]

**Pull request merged**